firstvoices/backend/tests/factories.py

```diff
@@ -178,6 +178,12 @@ def get_site_with_member(site_visibility, user_role):
     return site, user
 
 
+def get_site_with_user_as_member(site_visibility, user, user_role):
```

Comment:
Would the existing get_site_with_member work for this same case? It returns a site and user

Reply:
I think in this case it wouldn't work as some of the tests need a single user with membership to multiple sites. The `get_site_with_member` function creates a separate user with each call.

I have added an optional user parameter to the `get_site_with_member` function to add this functionality, and I have removed `get_site_with_user_as_member`.